lib/createlocaltracks.ts

```diff
@@ -91,80 +96,89 @@ function createLocalTracks(options) {
     LocalVideoTrack,
     MediaStreamTrack,
     Log,
-    video: isAudioVideoAbsent
+    video: isAudioVideoAbsent,
   }, options);
 
   const logComponentName = `[createLocalTracks #${++createLocalTrackCalls}]`;
-  const logLevels = buildLogLevels(options.logLevel);
-  const log = new options.Log('default', logComponentName, logLevels, options.loggerName);
+  const logLevels = buildLogLevels(fullOptions.logLevel);
+  const log = new fullOptions.Log('default', logComponentName, logLevels, fullOptions.loggerName);
+
+  const localTrackOptions = Object.assign({ log }, fullOptions);
 
   // NOTE(mmalavalli): The Room "name" in "options" was being used
   // as the LocalTrack name in asLocalTrack(). So we pass a copy of
   // "options" without the "name".
-  const localTrackOptions = Object.assign({ log }, options);
-  delete localTrackOptions.name;
+  // NOTE(joma): CreateLocalTracksOptions type does not really have a "name" property when used publicly by customers.
+  // But we are passing this property when used internally by other JS files.
+  // We can update this "any" type once those JS files are converted to TS.
+  delete (localTrackOptions as any).name;
 
-  if (options.audio === false && options.video === false) {
+  if (fullOptions.audio === false && fullOptions.video === false) {
     log.info('Neither audio nor video requested, so returning empty LocalTracks');
-    return Promise.resolve([]);
+    return [];
   }
 
-  if (options.tracks) {
+  if (fullOptions.tracks) {
     log.info('Adding user-provided LocalTracks');
-    log.debug('LocalTracks:', options.tracks);
-    return Promise.resolve(options.tracks);
+    log.debug('LocalTracks:', fullOptions.tracks);
+    return fullOptions.tracks;
   }
 
-  const extraLocalTrackOptions = {
-    audio: options.audio && options.audio.name
-      ? { name: options.audio.name }
+  const extraLocalTrackOptions: { audio: ExtraLocalTrackOption; video: ExtraLocalTrackOption; } = {
```

Comment:
Instead of defining the type inline, please create `ExtraLocalTrackOptions` type above (line 25)
```ts
type ExtraLocalTrackOptions = { audio: ExtraLocalTrackOption; video: ExtraLocalTrackOption; };
```

Then in here, you can do
```ts
const extraLocalTrackOptions: ExtraLocalTrackOptions = {...
```